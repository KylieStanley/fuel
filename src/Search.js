import React, { Component } from 'react'
import './styles/main.scss';


export default class Search extends Component {
	constructor() {
		super()
		this.state = {
			search: ''
		}
	}

	handleSearch = (e) => {
		this.setState({ search: e.target.value })
	}

	render() {
		return (
		<form onSubmit={this.handleSubmit}>
			<input type="text" placeholder="Search for Recipes" value={this.state.search} onChange={this.handleSearch} />
			<button className="search-btn"><i className="fas fa-search"></i></button>
		</form>
		)
	}
}