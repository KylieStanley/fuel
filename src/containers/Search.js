import React, { Component } from 'react'
import '../styles/main.scss';
import { connect } from 'react-redux'
import { cleanSearchString } from '../helper'
import { apiId, apiKey } from '../apiKey'
import { fetchRecipes } from '../thunks/fetchRecipes'


export class Search extends Component {
	constructor() {
		super()
		this.state = {
			search: ''
		}
	}

	handleChange = (e) => {
		this.setState({ search: e.target.value })
	}

	handleSubmit = (e) => {
		e.preventDefault()
		if (this.state.search) {
			const searchString = cleanSearchString(this.state.search)
			const url = `https://api.edamam.com/search?q=${searchString}&app_id=${apiId}&app_key=${apiKey}&from=0&to=30`
			this.props.fetchRecipes(url)
		}
	}

	render() {
		return (
		<form onSubmit={this.handleSubmit}>
			<input type="text" placeholder="Search for Recipes" value={this.state.search} onChange={this.handleChange} />
			<button className="search-btn"><i className="fas fa-search"></i></button>
		</form>
		)
	}
}

export const mapDispatchToProps = (dispatch) => ({
	fetchRecipes: (url) => dispatch(fetchRecipes(url))
})


export default connect(null, mapDispatchToProps)(Search)