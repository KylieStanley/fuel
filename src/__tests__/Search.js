import React from 'react'
import { shallow } from 'enzyme'
import Search from '../Search'

describe('Search', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<Search />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('simulates text change in input box',() => {
    const mockData = { target: {value: 'Chicken'}}
    const spy = jest.spyOn(wrapper.instance(), 'handleSearch')
    wrapper.instance().forceUpdate();
    const input = wrapper.find('input');
    input.simulate('change', mockData);

    expect(spy).toHaveBeenCalled()
  });
})
