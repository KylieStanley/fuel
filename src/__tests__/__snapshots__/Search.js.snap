// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          onChange={[Function]}
          placeholder="Search for Recipes"
          type="text"
          value=""
        />,
        <button
          className="search-btn"
        >
          <i
            className="fas fa-search"
          />
        </button>,
      ],
      "onSubmit": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onChange": [Function],
          "placeholder": "Search for Recipes",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <i
            className="fas fa-search"
          />,
          "className": "search-btn",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fas fa-search",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            onChange={[Function]}
            placeholder="Search for Recipes"
            type="text"
            value=""
          />,
          <button
            className="search-btn"
          >
            <i
              className="fas fa-search"
            />
          </button>,
        ],
        "onSubmit": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "placeholder": "Search for Recipes",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <i
              className="fas fa-search"
            />,
            "className": "search-btn",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fas fa-search",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
